spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/event/asset/AssetDeleted.java

```diff
@@ -9,25 +9,25 @@
  *
  *  Contributors:
  *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *       Fraunhofer Institute for Software and Systems Engineering - expending Event classes
  *
  */
 
 package org.eclipse.dataspaceconnector.spi.event.asset;
 
 import org.eclipse.dataspaceconnector.spi.event.Event;
-import org.eclipse.dataspaceconnector.spi.event.EventPayload;
 
 import java.util.Objects;
 
 /**
  * Describe an Asset deletion, after this has emitted, the Asset represented by the id won't be available anymore.
  */
-public class AssetDeleted extends Event<AssetDeleted.Payload> {
+public class AssetDeleted extends AssetEvent {
```

Comment:
```suggestion
public class AssetDeleted extends Event<AssetDeleted.Payload> {
```

I think the AssetDeleted should be related to its own payload, currently it's the same for both the events, but this could change in any moment

Reply:
If i change it like the way you suggest their is no superclass again.

But you are right i messed up the Relation. I want sth. like EventPayload <- AssetPayload <- AssetCreatedPayload/AssetDeleted Payload. I need to change it this way.

Reply:
Looking it from another perspective seems like the hierarchy is more related to the payload than to the event class itself... 
Maybe the only class we need to introduce is a `AssetEventPayload` that contains the `assetId` and that it could be used to filter out events.

Reply:
But i think that wont work.

We then their exist two Event Class hierarchies (Event and AssetDeleted/Created) and three  Payload Class hierarchies (EventPayload and AssetEventPayload and AssetCreatedPayload/AssetDeletedPayload)

Event -> EventPayload
AssetCreated/AssetDeleted -> AssetCreatedPayload/AssetDeletedPayload
? -> AssetEventPayload

So i think we need to create AssetEvent class to replace ?.

And the Inheritance should be this.

Event->AssetEvent->AssetCreated/AssetDeleted
EventPayload->AssetEventPayload->AssetCreatedPayload/AssetDeletedPayload

Mabey i dont see your idea.

But i definitly messed up the hierarchie in my code what needs to be fixed by me.